lib/compress.js

```diff
@@ -706,9 +706,9 @@ merge(Compressor.prototype, {
         return x;
     };
 
-    function is_iife_call(node) {
+    function is_iife_call(node, traditional) {
         if (node instanceof AST_Call && !(node instanceof AST_New)) {
-            return is_func_expr(node.expression) || is_iife_call(node.expression);
+            return is_func_expr(node.expression, traditional) || is_iife_call(node.expression);
```

Comment:
Tests fail if that change is made since `is_iife_call` is recursive. The inner IIFEs can be either traditional functions or arrow functions.